src/Symfony/Component/Form/Form.php

```diff
@@ -532,6 +532,8 @@ public function submit($submittedData, $clearMissing = true)
             $submittedData = null;
         } elseif (is_scalar($submittedData)) {
             $submittedData = (string) $submittedData;
+        } elseif ($this->config->getOption('allow_file_upload')) {
```

Comment:
@nicolas-grekas @HeahDude I need your help here. If I don't miss anything, we need to add this back as otherwise the fix introduced in #29307 will prevent file uploads using the `NativeRequestHandler` from being valid. Do you agree with this reasoning?

Reply:
Should we be even more strict and consider that if `allow_file_upload` is set, the value must be a file upload?

Reply:
I am not sure. I think there is some additional error handling if the the submitted data is not an uploaded file. But I am going to check that.

Reply:
If I don't miss anything, such a check would fail for example for a `FileType` where `multiple` is set to `true`.